src/f32/funcs.rs

```diff
@@ -139,6 +197,64 @@ pub(crate) mod sse2 {
         _mm_or_ps(result, _mm_castsi128_ps(test))
     }
 
+    /// Returns a vector whose components are the corresponding components of Angles modulo 2PI.
+    #[allow(dead_code)]
+    pub(crate) unsafe fn m128_mod_angles(angles: __m128) -> __m128 {
+        // From DirectXMath: XMVectorModAngles
+        let v = _mm_mul_ps(angles, XM_RECIPROCAL_TWO_PI.m128);
+        let v = m128_round(v);
+        XM_FMADD_PS!(XM_TWO_PI.m128, v, angles)
+    }
+
+    pub(crate) unsafe fn m128_sin(v: __m128) -> __m128 {
+        // From DirectXMath: XMVectorSin
+        //
+        // 11-degree minimax approximation
+        //
+        // Note that XMVectorSin clamps the value here with XMVectorModAngles,
+        // but this adds about 25% overhead.
+        //
+        // // Force the value within the bounds of pi
+        // let mut x = m128_mod_angles(v);
```

Comment:
Curious why this was commented out? From my experience with the approximation I was using here which I commented out, accuracy started to get a large error and angles grew beyond PI.